openrtb_ext/request.go

```diff
@@ -312,3 +344,116 @@ func (m ExtMultiBid) String() string {
+			return adjArray
+		}
+	}
+	if bidAdj.MediaType.Video != nil && bidType == BidTypeVideo {
+		if adjArray := getAdjustmentArrayForMediaType(bidAdj.MediaType.Video, bidderName.String(), dealID); adjArray != nil {
+			return adjArray
+		}
+	}
+	if bidAdj.MediaType.Audio != nil && bidType == BidTypeAudio {
+		if adjArray := getAdjustmentArrayForMediaType(bidAdj.MediaType.Audio, bidderName.String(), dealID); adjArray != nil {
+			return adjArray
+		}
+
+	}
+	if bidAdj.MediaType.Native != nil && bidType == BidTypeNative {
+		if adjArray := getAdjustmentArrayForMediaType(bidAdj.MediaType.Native, bidderName.String(), dealID); adjArray != nil {
+			return adjArray
+		}
+	}
+	if bidAdj.MediaType.WildCard != nil {
+		if adjArray := getAdjustmentArrayForMediaType(bidAdj.MediaType.WildCard, bidderName.String(), dealID); adjArray != nil {
+			return adjArray
+		}
+	}
+	return nil
+}
+
+// Priority For Returning Adjustment Array Based on Passed BidderName and DealID
+// #1: Are able to match bidderName and dealID
+// #2: Are able to match bidderName and dealID field is WildCard
+// #3: Bidder field is WildCard and are able to match DealID
+// #4: Wildcard bidder and wildcard dealID
+func getAdjustmentArrayForMediaType(bidAdjMap map[string]map[string][]Adjustments, bidderName string, dealID string) []Adjustments {
+	if _, ok := bidAdjMap[bidderName]; ok {
+		if _, ok := bidAdjMap[bidderName][dealID]; ok {
+			return bidAdjMap[bidderName][dealID]
+		} else if _, ok := bidAdjMap[bidderName][AdjWildCard]; ok {
+			return bidAdjMap[bidderName][AdjWildCard]
+		}
+	} else if _, ok := bidAdjMap[AdjWildCard]; ok {
+		if _, ok := bidAdjMap[AdjWildCard][dealID]; ok {
+			return bidAdjMap[AdjWildCard][dealID]
+		} else if _, ok := bidAdjMap[AdjWildCard][AdjWildCard]; ok {
+			return bidAdjMap[AdjWildCard][AdjWildCard]
+		}
+	}
+	return nil
+}
+
+func (bidAdjustments *ExtRequestPrebidBidAdjustments) ValidateBidAdjustments() bool {
+	if bidAdjustments == nil {
+		return true
+	}
+	if bidAdjustments.MediaType.Banner != nil {
+		if valid := findAndValidateAdjustment(bidAdjustments.MediaType.Banner); !valid {
+			return false
+		}
+	}
+	if bidAdjustments.MediaType.Video != nil {
+		if valid := findAndValidateAdjustment(bidAdjustments.MediaType.Video); !valid {
+			return false
+		}
+	}
+	if bidAdjustments.MediaType.Audio != nil {
+		if valid := findAndValidateAdjustment(bidAdjustments.MediaType.Audio); !valid {
+			return false
+		}
+	}
+	if bidAdjustments.MediaType.Native != nil {
+		if valid := findAndValidateAdjustment(bidAdjustments.MediaType.Native); !valid {
+			return false
+		}
+	}
+	return true
+}
+
+func findAndValidateAdjustment(bidAdjMap map[string]map[string][]Adjustments) bool {
+	for bidderName := range bidAdjMap {
+		for dealId := range bidAdjMap[bidderName] {
+			for _, adjustment := range bidAdjMap[bidderName][dealId] {
+				if !validateAdjustment(adjustment) {
+					return false
+				}
```

Comment:
I can't return immediately here because there could still be more adjustments that need to be validated. So I'd only return immediately if I get an invalid adjustment.